connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -1,27 +1,146 @@
+import assert from "node:assert";
+
+import type { ModelId } from "@dust-tt/types";
 import type { Client } from "node-zendesk";
 import { createClient } from "node-zendesk";
 
+import type { ZendeskFetchedArticle } from "@connectors/connectors/zendesk/lib/node-zendesk-types";
+import { ExternalOAuthTokenError } from "@connectors/lib/error";
+import { ZendeskBrandResource } from "@connectors/resources/zendesk_resources";
+
 export function createZendeskClient({
-  token,
+  accessToken,
   subdomain,
 }: {
-  token: string;
+  accessToken: string;
   subdomain: string;
 }) {
-  return createClient({ oauth: true, token, subdomain });
+  return createClient({ oauth: true, token: accessToken, subdomain });
 }
 
-export async function changeZendeskClientSubdomain({
-  client,
-  brandId,
-}: {
-  client: Client;
-  brandId: number;
-}) {
-  // TODO(2024-10-29 aubin): in some cases the brand in db is available and can be used to retrieve the subdomain
+/**
+ * Returns a Zendesk client with the subdomain set to the one in the brand.
+ * Retrieves the brand from the database if it exists, fetches it from the Zendesk API otherwise.
+ */
+export async function changeZendeskClientSubdomain(
+  client: Client,
+  {
+    connectorId = null,
```

Comment:
Why can connectorId be null?

Reply:
oops forgot to remove this (added this at first for an easier refactoring), will do thanks for catching 👍